### ISSUE TITLE
CI revamp.

### ISSUE THREAD

**Issue by JasonLovesDoggo** [elebumm/RedditVideoMakerBot#1973]
### Summary

A proper CI system that passes if a video is created and fails if not. Ideally, the video is also uploaded as a GH artifact.

### Is your feature request related to a problem?

Difficulty dealing with PR's

### Describe the solution you'd like

A proper CI system that passes if a video is created and fails if not. Ideally, the video is also uploaded as a GH artifact.

**Comment by OpenSourceSimon** [comment]
Yeah, would be great if we can add automatic testing! That would make PR merging so much easier. However, I asked Lewis a while ago if he could add the necessary environment variables to the repo, haven't heard back from him

**Comment by JasonLovesDoggo** [comment]
> Yeah, would be great if we can add automatic testing! That would make PR merging so much easier. However, I asked Lewis a while ago if he could add the necessary environment variables to the repo, haven't heard back from him

You create the tests. I'll worry about the env variables

**Comment by elebumm** [comment]
@OpenSourceSimon @JasonLovesDoggo 

Hit me up on discord! I can get these for you

**Comment by Popidge** [comment]
Will look into this for you, how exhaustive do you want the testing @JasonLovesDoggo? just one TTS provider or multiple?

**Comment by JasonLovesDoggo** [comment]
> Will look into this for you, how exhaustive do you want the testing [@JasonLovesDoggo](https://github.com/JasonLovesDoggo)? just one TTS provider or multiple?

That's more a testing suite question. You're free to pick your own scope
